server/src/main/java/org/opensearch/search/aggregations/InternalAggregation.java

```diff
@@ -112,13 +114,30 @@ public static ReduceContext forPartialReduction(
             ScriptService scriptService,
             Supplier<PipelineTree> pipelineTreeForBwcSerialization
         ) {
-            return new ReduceContext(bigArrays, scriptService, (s) -> {}, null, pipelineTreeForBwcSerialization);
+            return new ReduceContext(bigArrays, scriptService, (s) -> {}, null, pipelineTreeForBwcSerialization, null);
         }
 
         /**
          * Build a {@linkplain ReduceContext} to perform the final reduction.
          * @param pipelineTreeRoot The root of tree of pipeline aggregations for this request
          */
+        public static ReduceContext forFinalReduction(
+            BigArrays bigArrays,
+            ScriptService scriptService,
+            IntConsumer multiBucketConsumer,
+            PipelineTree pipelineTreeRoot,
+            CircuitBreaker breaker
+        ) {
+            return new ReduceContext(
+                bigArrays,
+                scriptService,
+                multiBucketConsumer,
+                requireNonNull(pipelineTreeRoot, "prefer EMPTY to null"),
+                () -> pipelineTreeRoot,
+                breaker
```

Comment:
I realize the multiBucketConsumer of reduce context already has breaker in it. Should we just use that?
https://github.com/opensearch-project/OpenSearch/blob/f7b31b05f0bfe2aec1ceac387329dc71dc14021a/server/src/main/java/org/opensearch/search/aggregations/MultiBucketConsumerService.java#L132